tide/master/ScreenControllerFactory.cpp

```diff
@@ -40,30 +40,69 @@
 #include "ScreenControllerFactory.h"
 
 #include "MultiScreenController.h"
-#include "PlanarController.h"
 
+#include <QMap>
 #include <QStringList>
 
+namespace
+{
+PlanarController::Type _getType(const QString& name)
+{
+    if (name == "UR9850")
+        return PlanarController::Type::TV_UR9850;
+    if (name == "UR9851")
+        return PlanarController::Type::TV_UR9851;
+    return PlanarController::Type::Matrix;
+}
+}
+
 std::unique_ptr<ScreenController> ScreenControllerFactory::create(
     const QString& ports)
 {
-    auto serialports = ports.split(';');
-    if (serialports.length() == 0)
+    const auto connections = parseInputString(ports);
+    if (connections.empty())
         return nullptr;
-    if (serialports.length() == 1)
-    {
-        return std::unique_ptr<ScreenController>(
-            new PlanarController(serialports[0],
-                                 PlanarController::Type::Matrix));
-    }
 
-    std::vector<std::unique_ptr<PlanarController>> controllers;
-    for (const auto serialport : serialports)
+    std::vector<std::unique_ptr<ScreenController>> controllers;
+    for (const auto& kv : connections.toStdMap())
     {
-        auto controller = std::unique_ptr<PlanarController>(
-            new PlanarController(serialport, PlanarController::Type::TV));
-        controllers.push_back(std::move(controller));
+        controllers.emplace_back(new PlanarController(kv.first, kv.second));
     }
+    if (controllers.size() == 1)
+        return std::move(controllers[0]);
+
     return std::unique_ptr<ScreenController>(
         new MultiScreenController(std::move(controllers)));
 }
+
+QMap<QString, PlanarController::Type> ScreenControllerFactory::parseInputString(
+    const QString& ports)
+{
+    QMap<QString, PlanarController::Type> map;
+
+    const auto connections = ports.split(';');
+    if (connections.empty())
+        return map;
+    for (const auto& connection : connections)
+    {
+        if (connection.isEmpty())
+            continue;
+
+        if (!connection.endsWith("#") && connection.contains("#"))
```

Comment:
to be reviewed, at least inverse the two conditions for readability. If someone inserts abc#def# this jumps to the second case, which is not the intention. I would still prefer a single code path with a split('#') at the beginning.